### PR TITLE
OSDOCS-8079: Port Architecture book to OSD/ROSA

### PR DESCRIPTION
**Pull request by bhardesty**
Version(s):
4.15+

Issue:
https://issues.redhat.com/browse/OSDOCS-8079

Link to docs preview:
* OCP: https://70602--ocpdocs-pr.netlify.app/openshift-enterprise/latest/architecture/
* OSD: https://70602--ocpdocs-pr.netlify.app/openshift-dedicated/latest/architecture/
* ROSA: https://70602--ocpdocs-pr.netlify.app/openshift-rosa/latest/architecture/

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

File: architecture/index.adoc [hunk 1]
```diff
@@ -76,8 +92,8 @@ The {product-title} installation program creates the Ignition configuration file
...
+endif::openshift-dedicated,openshift-rosa[]
 
 [id="about-admission-plug-ins"]
 == About admission plugins
-You can use xref:../architecture/admission-plug-ins.adoc#admission-plug-ins[admission plugins] to regulate how {product-title} functions. After a resource request is authenticated and authorized, admission plugins intercept the resource request to the master API to validate resource requests and to ensure that scaling policies are adhered to.
-Admission plugins are used to enforce security policies, resource limitations, or configuration requirements.
+You can use xref:../architecture/admission-plug-ins.adoc#admission-plug-ins[admission plugins] to regulate how {product-title} functions. After a resource request is authenticated and authorized, admission plugins intercept the resource request to the master API to validate resource requests and to ensure that scaling policies are adhered to. Admission plugins are used to enforce security policies, resource limitations, or configuration requirements.
```
Comment (dpuniaredhat) [inline architecture/index.adoc:99]: How about below, WDYT
```suggestion
Admission plugins can be utilized to regulate the functionality of {product-title}. These plugins intercept resource requests to the master API after authentication and authorization, validating them and ensuring adherence to scaling policies. They enforce security policies, resource limitations, and configuration requirements. For more information, see xref:../architecture/admission-plug-ins.adoc#admission-plug-ins[admission plugins].
```

File: modules/openshift-architecture-common-terms.adoc [hunk 2]
```diff
@@ -9,13 +9,20 @@
...
 authentication::
-To control access to an {product-title} cluster, a cluster administrator can configure user authentication and ensure only approved users access the cluster. To interact with an {product-title} cluster, you must authenticate to the {product-title} API. You can authenticate by providing an OAuth access token or an X.509 client certificate in your requests to the {product-title} API.
+To control access to an {product-title} cluster,
+ifndef::openshift-dedicated,openshift-rosa[]
+a cluster administrator
+endif::openshift-dedicated,openshift-rosa[]
+ifdef::openshift-dedicated,openshift-rosa[]
+an administrator with the `dedicated-admin` role
+endif::openshift-dedicated,openshift-rosa[]
+can configure user authentication and ensure only approved users access the cluster. To interact with an {product-title} cluster, you must authenticate to the {product-title} API. You can authenticate by providing an OAuth access token or an X.509 client certificate in your requests to the {product-title} API.
```
Comment (dpuniaredhat) [inline modules/openshift-architecture-common-terms.adoc:25]:
```suggestion
can configure user authentication to ensure only approved users access the cluster. To interact with an {product-title} cluster, you must authenticate with the {product-title} API. Authentication can be achieved by providing an OAuth access token or an X.509 client certificate in your requests to the {product-title} API.
```

File: architecture/index.adoc (hunk 1 above)
Comment (wangke19) [reply architecture/index.adoc:99]: There are more than 30 admission controller plugins shipped with Kubernetes, so suggested change as below,
```suggestion
You can use xref:../architecture/admission-plug-ins.adoc#admission-plug-ins[admission plugins] to regulate how {product-title} functions. After a resource request is authenticated and authorized, admission plugins intercept the resource request to the master API to validate resource requests and to ensure that scaling policies are adhered to. Admission plugins are used to enforce security policies, resource limitations, or configuration requirements and other settings.
```

File: modules/sd-vs-ocp.adoc [hunk 3]
```diff
@@ -16,24 +16,42 @@ Review the following differences between {product-title} and a standard installa
...
 
-|You set up control plane components such as the API server and etcd on machines that get the `control` role. You can modify the control plane components, but keep in mind that you are responsible for backing up, restoring, and making control plane data highly available.
+|You set up control plane components such as the API server and etcd on machines that get the `control` role. You can modify the control plane components, but are responsible for backing up, restoring, and making control plane data highly available.
 |Red Hat sets up the control plane and manages the control plane components for you. The control plane is highly available.
 
 |You are responsible for updating the underlying infrastructure for the control plane and worker nodes. You can use the OpenShift web console to update {OCP} versions.
-|Red Hat automatically notifies the customer when updates are available. You can manually or automatically schedule upgrades in {cluster-manager-first}.
+|Red Hat automatically notifies the customer when updates are available. You can manually or automatically schedule upgrades in {cluster-manager}.
```
Comment (skopacz1) [inline modules/sd-vs-ocp.adoc:52]: Update is the preferred term per our [term glossary](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#update):
```suggestion
|Red Hat automatically notifies the customer when updates are available. You can manually or automatically schedule updates in {cluster-manager}.
```

File: osd_getting_started/osd-getting-started.adoc [hunk 4]
```diff
@@ -12,7 +12,7 @@ Follow this getting started document to quickly create a {product-title} cluster
 [id="osd-getting-started-prerequisites"]
 == Prerequisites
 
-* You reviewed the xref:../osd_architecture/osd-understanding.adoc#osd-understanding[introduction to {product-title}] and the documentation on xref:../osd_architecture/osd-architecture.adoc#osd-architecture[architecture concepts].
+* You reviewed the xref:../osd_architecture/osd-understanding.adoc#osd-understanding[introduction to {product-title}] and the documentation on xref:../architecture/index.adoc[architecture concepts].
```
Comment (skopacz1) [inline osd_getting_started/osd-getting-started.adoc:15]: All xrefs need an anchor ID even if they link to the top of the assembly, [per our guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#links-hyperlinks-and-cross-references):
```suggestion
* You reviewed the xref:../osd_architecture/osd-understanding.adoc#osd-understanding[introduction to {product-title}] and the documentation on xref:../architecture/index.adoc#architecture-overview[architecture concepts].
```

File: osd_install_access_delete_cluster/creating-a-gcp-cluster.adoc [hunk 5]
```diff
@@ -12,7 +12,7 @@ You can install {product-title} on {GCP} by using your own GCP account through t
 [id="osd-creating-a-cluster-on-gcp-prerequisites_{context}"]
 == Prerequisites
 
-* You reviewed the xref:../osd_architecture/osd-understanding.adoc#osd-understanding[introduction to {product-title}] and the documentation on xref:../osd_architecture/osd-architecture.adoc#osd-architecture[architecture concepts].
+* You reviewed the xref:../osd_architecture/osd-understanding.adoc#osd-understanding[introduction to {product-title}] and the documentation on xref:../architecture/index.adoc[architecture concepts].
```
Comment (skopacz1) [inline osd_install_access_delete_cluster/creating-a-gcp-cluster.adoc:15]: Same comment as above:
```suggestion
* You reviewed the xref:../osd_architecture/osd-understanding.adoc#osd-understanding[introduction to {product-title}] and the documentation on xref:../architecture/index.adoc#architecture-overview[architecture concepts].
```

File: osd_install_access_delete_cluster/creating-an-aws-cluster.adoc [hunk 6]
```diff
@@ -12,7 +12,7 @@ You can install {product-title} on {AWS} by using your own AWS account through t
 [id="osd-creating-a-cluster-on-aws-prerequisites_{context}"]
 == Prerequisites
 
-* You reviewed the xref:../osd_architecture/osd-understanding.adoc#osd-understanding[introduction to {product-title}] and the documentation on xref:../osd_architecture/osd-architecture.adoc#osd-architecture[architecture concepts].
+* You reviewed the xref:../osd_architecture/osd-understanding.adoc#osd-understanding[introduction to {product-title}] and the documentation on xref:../architecture/index.adoc[architecture concepts].
```
Comment (skopacz1) [inline osd_install_access_delete_cluster/creating-an-aws-cluster.adoc:15]: Same comment as above:
```suggestion
* You reviewed the xref:../osd_architecture/osd-understanding.adoc#osd-understanding[introduction to {product-title}] and the documentation on xref:../architecture/index.adoc#architecture-overview[architecture concepts].
```

File: rosa_hcp/rosa-hcp-sts-creating-a-cluster-quickly.adoc [hunk 7]
```diff
@@ -26,7 +26,7 @@ Since it is not possible to upgrade or convert existing ROSA clusters to a {hcp}
 ====
 
 .Further reading
-* For a comparison between {hcp-title} and ROSA Classic, see the xref:../rosa_architecture/rosa_architecture_sub/rosa-architecture-models.adoc#rosa-hcp-classic-comparison_rosa-architecture-models[Comparing architecture models] documentation.
+* For a comparison between {hcp-title} and ROSA Classic, see the xref:../architecture/rosa-architecture-models.adoc#rosa-hcp-classic-comparison_rosa-architecture-models[Comparing architecture models] documentation.
 * See the AWS documentation for information about link:https://docs.aws.amazon.com/rosa/latest/userguide/getting-started-hcp.html[Getting started with ROSA with HCP using the ROSA CLI in auto mode].
 
 .Additional resources
```
Comment (skopacz1) [inline rosa_hcp/rosa-hcp-sts-creating-a-cluster-quickly.adoc:32]: This is out of scope but if you'd like to fix it here, this section needs a role tag per our [guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#additional-resources-sections):
```suggestion
[role="_additional-resources"]
.Additional resources
```

**Review by skopacz1 — COMMENTED** [review]
Left some comments but otherwise looks good to me! Excellent work with this very thorough and diligent PR, let me know if you have questions about any of my feedback.

File: modules/nvidia-gpu-csps.adoc [hunk 8]
```diff
@@ -4,15 +4,27 @@
 
 :_mod-docs-content-type: CONCEPT
 [id="nvidia-gpu-csps_{context}"]
+ifndef::openshift-dedicated,openshift-rosa[]
 = GPUs and CSPs
+endif::openshift-dedicated,openshift-rosa[]
+ifdef::openshift-dedicated,openshift-rosa[]
+= GPUs and ROSA
+endif::openshift-dedicated,openshift-rosa[]
```
[the remark this answers is not in the document]
Comment (bhardesty) [reply modules/nvidia-gpu-csps.adoc:12]: Whoops - good catch!

**Comment by bscott-rh** [comment]
Merge review LGTM.

**Comment by bscott-rh** [comment]
Cherrypicks will be merged once the Prow CI issues are resolved.

**Comment by bscott-rh** [comment]
Cherrypicks have been merged to 4.15 and 4.16

**Pull request merged**